lightning-background-processor/src/lib.rs

```diff
@@ -744,6 +768,38 @@ mod tests {
 		assert!(bg_processor.stop().is_ok());
 	}
 
+	#[test]
+	fn test_not_pruning_network_graph_until_graph_sync_completion() {
+		let mut nodes = create_nodes(2, "test_not_pruning_network_graph_until_graph_sync_completion".to_string());
+		let channel_value = 100000;
+		let data_dir = nodes[0].persister.get_data_dir();
+		let (sender, receiver) = std::sync::mpsc::sync_channel(1);
+		let persister = Persister::new(data_dir.clone()).with_graph_sender(sender);
+		let network_graph = nodes[0].network_graph.clone();
+		let features = ChannelFeatures::empty();
+		network_graph.update_channel_from_partial_announcement(42, 53, features, nodes[0].node.get_our_node_id(), nodes[1].node.get_our_node_id());
+		let original_graph_description = network_graph.to_string();
+		assert!(original_graph_description.contains("42: features: 0000, node_one:"));
+
+		let event_handler = |_: &_| {};
+		let bg_processor = BackgroundProcessor::start(persister, event_handler, nodes[0].chain_monitor.clone(), nodes[0].node.clone(), nodes[0].net_graph_msg_handler.clone(), nodes[0].peer_manager.clone(), true, nodes[0].logger.clone());
+
+		let reception_result = receiver
+			.recv_timeout(Duration::from_secs(EVENT_DEADLINE));
+		assert!(reception_result.is_err());
```

Comment:
Hmm... this is a 5-second sleep in practice. Seems like you need at least a second for the prune logic, though. 😕 Ideally, we could use the `Time` trait being refactored out in #1418. But even then you can't be guaranteed that the code has been executed yet in the other thread. May have to add logging and check for it similarly as done in `test_timer_tick_called`.

Reply:
Probably something for a future PR, but should these loop-based unit tests perhaps have a timeout exit condition to avoid potentially stalling a failing test indefinitely?

Reply:
> Probably something for a future PR,

Just replacing this with a loop like `test_timer_tick_called` should be pretty trivial, no? The larger refactor can wait for a future PR, of course, but no need adding 5 seconds to our tests.

>but should these loop-based unit tests perhaps have a timeout exit condition to avoid potentially stalling a failing test indefinitely?

Yeah, definitely!

Reply:
> Just replacing this with a loop like `test_timer_tick_called` should be pretty trivial, no? The larger refactor can wait for a future PR, of course, but no need adding 5 seconds to our tests.
> 

Ah, nevermind. Just saw you already added the loop. Thanks!